fix: empty datatable

andriacap opened the pull request:
Pb solved: datatable empty when "colsname" or "siteGroups" / "sites" are not both loaded .

Add *ngIf to wait those two inputs are loaded

Reviewed-by: andriac
[Refs_ticket]: #48

Closes: #48

andriacap committed:
fix: empty datatable

Pb solved: datatable empty when "colsname" or "siteGroups" / "sites"
are not both loaded .

Add *ngIf to wait those two inputs are loaded

Reviewed-by: andriac
[Refs_ticket]: #48

mvergez commented:
Je viens de tester et cela ne fonctionne pas avec le test ci-dessous : 
- Accueil GeoNature => Rechargement de la page
- Monitorings puis accès aux sites
- La table est vide

Je suis bien sur la branche : fix/display-datatable-infos et j'ai recompilé le front, sait-on jamais ^^

andriacap commented:
Merci pour ton retour oui effectivement ça ne fonctionnait pas dans ce cas .
J'ai ciblé le problème : 
- La fonction private `initConfig()` du service `api-geom.service` (voir ci dessous)  est éxécuté après le montage des composants ce qui fait qu'on a pas accès à toutes les données qui sont chargé dans cette fonction  (notamment this.colsname qui est toujours empty d'où notre problème)

https://github.com/NaturalSolutions/gn_module_monitoring/blob/d73b7ae9f2a9686cad11671e21cef3b7a6811777/frontend/app/services/api-geom.service.ts#L32-L62

Du coup pour le faire fonctionner , en attendant d'avoir une meilleure solution, j'ai rendu public la fonction et on l'appel au moment de l'initialisation du composant où on en a besoin
(exemple d'u gars qui parlait de ce problème : https://dev.to/slavius/wait-for-injectable-service-init-in-angular-38l8 ) 

Je vais pusher ce travail que j'ai fait

andriacap committed:
fix: change way to load colsname and init Object

In order to get all data needed I change the api-geom.service.ts to
allow using the initConfig() method.

Create Resolver to load needed data before to init component (it mays be
optional to use like this) (different way between sitegroup and site
component)

Reviewed-by: andriac
[Refs_ticket]: #48
